### ISSUE TITLE
Can't access xvg metadata

### ISSUE THREAD

**Issue by bk322** [Becksteinlab/GromacsWrapper#13]
Hi Oliver, today I tried to plot xvg data in matplotlib. I read it with GromacsWrapper -- but can't access xvg metadata:

```
@    title "Gromacs Energies"
@    xaxis  label "Time (ps)"
@    yaxis  label "(kJ/mol)"
@TYPE xy
@ view 0.15, 0.15, 0.75, 0.85
@ legend on
@ legend box on
@ legend loctype view
@ legend 0.78, 0.8
@ legend length 2
@ s0 legend "Potential"
```

I need xaxis, yaxis and legend in particular:

```
>>> from gromacs import fileformats
>>> Ener = fileformats.xvg.XVG('energy.xvg')
>>> Ener.metadata
{}
>>> Ener.names
[]
```

I'm using bleeding edge GromacsWrapper, I get a moderate issue on startup:

```
/home/boris/.local/lib/python2.7/site-packages/GromacsWrapper-0.3.1-py2.7.egg/gromacs/__init__.py:324: GromacsImportWarning: Some Gromacs commands were NOT found; maybe source GMXRC first? The following are missing:
['a_gridcalc', 'a_ri3dc', 'g_count', 'g_flux', 'g_pme_error', 'g_ri3dc']
```

I don't think it is relevant.

**Comment by ostueker** [comment]
It seems the XVG reader has finally learned to read metadata, names and axis-labels from Gromacs-Generated XVG files. :tada:

However the behaviour is somewhat inconsistent. See for example:

```python
In [1]: import gromacs.formats
   ...: print(gromacs.__version__)
0.8.2

In [2]: xvg = gromacs.formats.XVG(filename='examples/gmx_energy_temp.xvg')

In [3]: print(xvg.names)
   ...: print(xvg.metadata)
[]
{}

In [4]: print(xvg.array)
[[  0.         1.         2.         3.         4.         5.      ]
 [300.405334 301.206299 300.53598  301.529816 298.039734 300.182587]]

In [5]: print(xvg.names)
   ...: print(xvg.metadata)
['Temperature']
{'legend': ['on', 'box on', 'loctype view', '0.78, 0.8', 'length 2']}

In [6]: 
```
The first time I access the names and metadata attributes, they are empty, but once I have accessed the array once, they are suddenly populated. :thinking: 

Cheers,
ostueker (Oliver S.)

**Comment by orbeckst** [comment]
The XVGReader does lazy-loading, it only actually reads the file when the array is created.

That's admittedly a bit confusing, especially when it comes to the metadata.

Btw, I probably should have closed this issue when PR #18 was merged, which added some metadata parsing (although not all).

**Comment by orbeckst** [comment]
At least some of the metadata are read (added in PR #18). If anything else is missing that you'd really want then please open a new issue. Thank you!
